gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -146,12 +156,98 @@ protected void addLineageSourceInfo(SourceState sourceState, SourceEntity entity
 
   @Override
   protected List<WorkUnit> generateWorkUnits(SourceEntity sourceEntity, SourceState state, long previousWatermark) {
+    String partitionType = state.getProp(PARTITION_TYPE, "");
+    if (partitionType.equals("PK_CHUNKING")) {
+      return generateWorkUnitsPkChunking(sourceEntity, state, previousWatermark);
```

Comment:
This is where to comment the right use cases to enable `PK_CHUNKING`, as it's the entry point
```
      // pk-chunking only supports start-time by source.querybased.start.value, and does not support end-time.
      // always ingest data later than or equal source.querybased.start.value.
      // we should only pk chunking based work units only in case of snapshot/full ingestion
```

Reply:
Thanks, will add it!